activerecord/lib/active_record/connection_adapters/abstract/connection_handler.rb

```diff
@@ -174,8 +174,6 @@ def retrieve_connection(connection_name, role: ActiveRecord::Base.current_role,
         unless pool
           if shard != ActiveRecord::Base.default_shard
             message = "No connection pool for '#{connection_name}' found for the '#{shard}' shard."
-          elsif ActiveRecord::Base.connection_handler != ActiveRecord::Base.default_connection_handler
-            message = "No connection pool for '#{connection_name}' found for the '#{ActiveRecord::Base.current_role}' role."
```

Comment:
We no longer use multiple handlers in the pool so we don't need this error anymore.